File: DTOs/Enums/AppointmentStatus.swift
import Foundation

/// `AppointmentStatus` описывает состояние appoinment
///
/// Используется для понимание кто отклонил или подтвердил
/// Так же отображает состояние успешно ли завершился или провалился
public enum AppointmentStatus: String, Codable, Hashable, Equatable {

    /// Этот статус ожидает подтверждения либо перехода по ссылки аттача customer и approve
    case requested
    /// Те кто подтвердили запись: `customer_approved` или `master_approved` или `both_approved`
    /// Так же этот статус подойдёт для отображение того
    /// что запись завершена успешно если `approved_all` и время процедуры уже прошло
    case customerApproved = "customer_approved"
    case masterApproved = "master_approved"
    case bothApproved = "both_approved"
    /// Те кто отклонил запись: `customer_declined` или `master_declined`
    /// Так же этот статус подойдёт для отображение того
    /// что кто то не пришёл на процедуру и время процедуры уже прошло
    case customerDeclined = "customer_declined"
    case masterDeclined = "master_declined"
}

File: Node/src/dto/enums/AppointmentStatus.ts
///  Описывает состояние Appointment
///  Поле: AppointmentEmployee | AppointmentCustomer
///
/// - `BOTH_APPROVED`    : Все участники Appointment подтвердили запрос
/// - `CUSTOMER_DECLINED`: Customer отменил Appointment
/// - `MASTER_DECLINED`  : Какой-то Employee отменил Appointment
/// - `MASTER_APPROVED`  : (AppointmentEmployee) соответствующий Employee подтвердил участие
/// - `CUSTOMER_APPROVED`: (AppointmentCustomer) соответствующий Customer подтвердил участие
/// - `REQUESTED`        : соответствующий Employee/Customer не подтвердил свое участие
export enum AppointmentStatus {
    requested = "requested",
    customerApproved = "customer_approved",
    masterApproved = "master_approved",
    bothApproved = "both_approved",
    customerDeclined = "customer_declined",
    masterDeclined = "master_declined",
}